### DISCUSSION TITLE
Button group?

### DISCUSSION THREAD

**Discussion by dmenne** [Budibase/budibase#5237] [Help] [answered]
Spectrum has [button groups](https://spectrum.adobe.com/page/button-group/).  Is it possible to create these in Budibase?

**Comment by melohagan** [comment]
Hi @dmenne 

What's your use case?
You could put related buttons within a container in budibase.

**Comment by dmenne** [comment]
Putting into a container is the trivial part. 

Use case: Alternatives to what elsewhere is called "radio boxes" - Mutually exclusive selection with automatic deselection, and ONE database field with an enum connected. In practice (with patients not always used to pinpoint exactly) using buttons instead of the small circles helps a lot.

Bootstrap also has these. And see the link to Spectrum in the original message

**Comment by melohagan** [comment, marked as answer]
I don't believe there are any near term plans to add Spectrum button groups, however you can build button groups in budibase:

![button_group_demo](https://user-images.githubusercontent.com/101575380/161580719-daf58406-848e-4886-934b-dcb9536ddc70.gif)

This makes use of the *Show component condition* (https://docs.budibase.com/docs/conditions) as well as <a href="https://docs.budibase.com/docs/app-state">App State</a>

The layout has a container for each combination of buttons, and then shows the relevant container based on the state **ButtonGroupIndex**
If the state is **undefined** then it will pick the DB value from the repeater. 

![Screenshot 2022-04-04 at 16 43 48](https://user-images.githubusercontent.com/101575380/161581389-63c4798a-bb02-4eb3-b75f-ee8704ddab37.png)

I made good use of the duplicate functionality!
![Screenshot 2022-04-04 at 16 47 02](https://user-images.githubusercontent.com/101575380/161582023-451d05ed-51c7-445a-b1e9-d785b434d88c.png)

Here is the app export: 
[button_group_radio-export-1649086774543.txt](https://github.com/Budibase/budibase/files/8411136/button_group_radio-export-1649086774543.txt)

---

I know this isn't ideal, particularly for large button groups, but hopefully it's enough to get you started!

**Comment by dmenne** [comment]
Thanks  for the idea. Essentially it means that there is no "pick-one" beside combo boxes, which are not acceptable in our use case.
